common/djangoapps/student/views/login.py

```diff
@@ -386,6 +386,28 @@ def send_reactivation_email_for_user(user):
     return JsonResponse({"success": True})
 
 
+@ensure_csrf_cookie
+def verify_user_password(request):
```

Comment:
What if the user doesn't have an edX password?  For example, they could use a 3rd party SSO provider like Google or Facebook?

Have we thought about having an email confirmation process instead?

Reply:
We have talked about the issues with 3rd party auth with this. For the first iteration, we will ask learners to reset their password before they can go through this flow.